### PR TITLE
fix nightly.yaml

### PR DESCRIPTION
**Pull request by yikaiMeta**
Test plan:

### CODE REVIEW

**Comment by yikaiMeta** [comment]
the default Linux github-hosted runner has 2 GPU, 7GB memory and 14 GB storage.
Here is github's official doc: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners
Haven't found the explanation of linux.20_04.4x but lots of our github Action yaml files are using it.

**Comment by yikaiMeta** [comment]
Push nightly job runs all tests. Two tests fail.

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -22,7 +23,7 @@ jobs:
           pip install -e .[dev]
           pip install twine
       - name: Run tests
-        run: python -m unittest discover --verbose --start-directory . --pattern "*_test.py"
+        run: pytest --verbose -o python_files="*_test.py"
```
Comment (d4l3k) [inline .github/workflows/nightly.yaml:26]: The filter isn't necessary I don't think

Our main python tests just call `pytest`: https://github.com/pytorch/torchx/blob/9f20e5a55233288a3ba6fb9bec4e72d1e5fbf6c2/.github/workflows/python-unittests.yaml#L40

File: .github/workflows/nightly.yaml [hunk 2]
```diff
@@ -4,15 +4,16 @@ on:
   # run every day at 11:15am
   schedule:
     - cron:  '15 11 * * *'
+  pull_request:
```
Comment (d4l3k) [inline .github/workflows/nightly.yaml:7]: please remove before landing

**Review by d4l3k — APPROVED** [review]
generally looks good to me -- think the test failures are on main so that's expected

**Comment by ryxli** [comment]
@yikaiMeta 

Looks like nightly push runs still fail?
https://github.com/pytorch/torchx/actions/runs/8553750453/job/23437562048#step:5:13

Latest nightly release is from [2024.2.12](https://pypi.org/project/torchx-nightly/)

 
Believe `pytest-cov` needs to be added to [dev-requirements.txt](https://github.com/pytorch/torchx/blob/main/dev-requirements.txt)

**Pull request merged**